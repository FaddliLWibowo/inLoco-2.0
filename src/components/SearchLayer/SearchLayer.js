import React from 'react'

const SearchLayer = ({onKeyUpSearch, onBtnCleanSearch, searchString}) => {
    let input
    var searchIconClass = 'search-layer--icon fa fa-'
    var onClickEvent = null
    if (searchString !== undefined && searchString.length === 0) {
        searchIconClass += 'search'
    } else {
        searchIconClass += 'close'
        onClickEvent = () => {
            input.value = ''
            onBtnCleanSearch()
        }
    }
    return (
        <form action="#" className="search-layer">
            <label htmlFor="searchLayer" className="search-layer--title">
                Ou pesquise por aqui:
                <span className={searchIconClass} onClick={onClickEvent}></span>
                <input
                    type="text"
                    id="searchLayer"
                    ref={node => {input = node;}}
                    onKeyUp={() => {
                        onKeyUpSearch(input.value)
                    }}
                    className="search-layer--input"
                    placeholder="Ex.: Educação"/>
            </label>
        </form>
    )
}

export default SearchLayer
