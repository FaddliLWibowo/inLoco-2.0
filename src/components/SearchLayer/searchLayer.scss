.search-layer {
    background-color: $search-layer-bg-color;
    bottom: 0;
    font-family: sans-serif;
    padding: 15px 12px;
    position: fixed;
    left: -1px;

    &--title {
        color: $menu-text-color;
        display: block;
        font-size: 14px;
        position: relative;
    }

    &--icon {
        background-color: $search-layer-bg-color;
        bottom: 7px;
        font-size: 16px;
        position: absolute;
        right: 0;
        z-index: 1;
    }

    &--input {
        background: transparent;
        border: 0;
        border-bottom: 1px solid $menu-text-color;
        color: $search-layer-input-color;
        font-size: 12px;
        font-style: italic;
        margin-top: 10px;
        padding-bottom: 5px;
        width: 100%;

        &:focus {
            outline: 0;
        }

        &::placeholder {
            color: $search-layer-input-color;
        }
    }
}
