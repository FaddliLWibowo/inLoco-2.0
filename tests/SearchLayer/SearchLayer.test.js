import React from 'react';
import SearchLayer from '../../src/components/SearchLayer/SearchLayer.js';
import renderer from 'react-test-renderer';

it('SearchLayer component renders correctly', () => {
    const searchLayer  = shallow(<SearchLayer/>);
    expect(searchLayer).toMatchSnapshot();
});
