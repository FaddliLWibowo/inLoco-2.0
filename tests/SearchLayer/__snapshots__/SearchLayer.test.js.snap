// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchLayer component renders correctly 1`] = `
<form
  action="#"
  className="search-layer"
>
  <label
    className="search-layer--title"
    htmlFor="searchLayer"
  >
    Ou pesquise por aqui:
    <span
      className="search-layer--icon fa fa-search"
    />
    <input
      className="search-layer--input"
      id="searchLayer"
      onKeyUp={[Function]}
      placeholder="Ex.: IDEB, Escola, BatalhÃµes da PMERJ"
      type="text"
    />
  </label>
</form>
`;
